Test append env instead of override

jasnell opened the pull request:
Issue: https://github.com/nodejs/node/issues/25
From: https://github.com/joyent/node/commit/e64ee2b3f7b4067101b0291f1add842353cd6865

Original commit message:

```
Some tests that rely on some environment variables being passed to child
processes would fail because they reset the child processes'
environement instead of appending to it. This would break on test
environments where some custom environment variables are needed to make
node work properly.
```

This port is modified to move the function into common.js per
https://github.com/nodejs/node/issues/25#issuecomment-104804881

jasnell commented:
Ugh.. the readme commit got bundled into this by mistake... hold on, we redo it

jasnell commented:
/cc @Fishrock123 @rvagg

Fishrock123 commented:
@jasnell it *may* be better to target these onto `master-with-merges` and then cherry-pick to master.. maybe? I'm not 100% sure.

Fishrock123 commented:
I would structure the final commit log as this, but discuss:

```
Some tests that rely on some environment variables being passed to child
processes would fail because they reset the child processes'
environment instead of appending to it. This would break on test
environments where some custom environment variables are needed to make
node work properly.

PORT-FROM: joyent/node @ e64ee2b3f7b4067101b0291f1add842353cd6865
This port is modified to move the function into common.js per
https://github.com/nodejs/node/issues/25#issuecomment-104804881

PR-URL: https://github.com/nodejs/node/pull/43
<Reviewed-By:>
```

I'm not sure I would separate out the original commit message as being original?

rvagg commented on test/sequential/test-net-GH-5504.js:
```diff
@@ -53,10 +53,10 @@ function parent() {
-    env: {
+    env: common.extendEnv({
       NODE_DEBUG: 'net',
       NODE_COMMON_PORT: process.env.NODE_COMMON_PORT,
```
redundant

rvagg commented on test/sequential/test-util-debug.js:
```diff
@@ -31,7 +31,10 @@ function test(environ, shouldWrite) {
-    env: { NODE_DEBUG: environ, HOME: process.env.HOME }
+    env: common.extendEnv({
+      NODE_DEBUG: environ,
+      HOME: process.env.HOME
```
redundant

rvagg commented:
lgtm

jasnell commented:
@rvagg ... removed the redundant environment variables.
@Fishrock123 ... let me know if it looks good to you

Fishrock123 commented:
Actual patches LGTM, if you prepare the commit message (and / or squash) I can just pull the `.patch` off of this and land it.

jasnell committed:
tests: append instead of override environment

From: https://github.com/joyent/node/commit/e64ee2b3f7b4067101b0291f1add842353cd6865

Original commit message:
```
Some tests that rely on some environment variables being passed to child
processes would fail because they reset the child processes'
environement instead of appending to it. This would break on test
environments where some custom environment variables are needed to make
node work properly.
```

This port is modified to move the function into common.js per
https://github.com/nodejs/node/issues/25#issuecomment-104804881

jasnell commented:
@Fishrock123 ... done!

Fishrock123 commented:
My bad for not doing this sooner. Landed in c76c40c5ee1e3e5db69cf73ffd3cde3b1353f0c3!

(And master, but that is slightly harder to keep track of.)
